app/screens/phone-auth-screen/request-phone-code-login.ts

```diff
@@ -264,8 +229,52 @@ export const useRequestPhoneCodeLogin = (): UseRequestPhoneCodeReturn => {
           setStatus(RequestPhoneCodeStatus.Error)
           setError(ErrorType.RequestCodeError)
         }
+      } else {
+        // we first register captcha, which will set geetestValidationData
+        registerCaptcha()
       }
-    })()
+    }
+
+    const appCheckPath = async () => {
+      try {
+        await axios.post(
+          authUrl + "/auth/phone/code-appcheck",
+          {
+            phone: validatedPhoneNumber,
+            channel: phoneCodeChannel,
+          },
+          {
+            headers: {
+              Appcheck: appCheckToken,
+            },
+          },
+        )
+
+        setStatus(RequestPhoneCodeStatus.SuccessRequestingCode)
+      } catch (err) {
+        setStatus(RequestPhoneCodeStatus.Error)
+        if (
+          isAxiosError(err) &&
+          err.response?.data.error === "UserCodeAttemptIdentifierRateLimiterExceededError"
+        ) {
+          setError(ErrorType.TooManyAttemptsError)
+        } else {
+          setError(ErrorType.RequestCodeError)
+        }
+      }
+    }
+
+    const skipRequestPhoneCode =
+      appConfig.galoyInstance.name === "Local" ||
+      appConfig.galoyInstance.name === "Staging"
+
+    if (skipRequestPhoneCode) {
```

Comment:
I think these functions are being called multiple times due to them being in a useEffect. Can you create a function that handles splitting along the paths and then call it after line 331 when the phone number is submitted. That would be similar to the previous implementation.

Reply:
you would like `captchaPath` and `appCheckPath` to be outside the useEffect?

Reply:
Yes. It seems like the trigger for the paths is when a user submits their phone number, so I think it is better if it is explicitly called as a function during submitPhoneNumber instead of called with a useEffect which runs when dependencies change.

Reply:
So actually the fact it's been called by useEffect and not directly is currently on purpose. when `submitPhoneNumber ` (renamed to userSubmitPhoneNumber) is called, it needs to go through captcha for the normal flow, which then asynchronously update the dependency when captcha is called. 

It's quite a convoluted logic, and I actually have added some comment to try to clarify it, but this logic is not something this PR is changing, it was already like this before. 

Maybe a better design would be to go with a proper/tested "state flow management" with a `useReducer` like we have in the send bitcoin destination flow, but I would not that as part of this PR

We will want to replace the captcha library imo so maybe would be a good time to do that at this time?

Reply:
That makes sense. I guess the only potential change would be to keep `registerCaptcha` as part of `userSubmitPhoneNumber` under an if statement to make the the captchaPath simpler. Up to you if you think that would improve it.

Reply:
I would want to look at it more when switching the captcha library, most of the current would then be removed